### PR TITLE
Chokidar instead of fs watch

### PR DESCRIPTION
**Pull request by coffee-cup**
Use [chokidar](https://github.com/paulmillr/chokidar) instead of fs.watch because chokidar

### CODE REVIEW

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export const start = async (
         );
       })
       .on("error", e => {
-        if (portNumber - startingPort > portTriesLimit) {
+        if (portNumber - port > portTriesLimit) {
```
Comment (SamirTalwar) [inline packages/server/src/index.ts]: This seems unrelated.

File: packages/server/src/index.ts [hunk 2]
```diff
@@ -60,17 +61,16 @@ export const start = async (
 
   app.use(Express.static(outDir));
 
-  fs.watch(process.cwd(), {recursive: true}, async (_, filename) => {
-    // TODO: Try/catch?
```
Comment (SamirTalwar) [inline packages/server/src/index.ts]: Does chokidar handle async functions well?

Comment (coffee-cup) [reply packages/server/src/index.ts]: it doesn't do anything with return value so I don't think it matters. It just expects a function with signature `(string, fs.Stats) => void`

File: packages/server/src/index.ts (hunk 1 above)
Comment (coffee-cup) [reply packages/server/src/index.ts]: You are correct, I'll put this in another PR

File: packages/server/src/index.ts (hunk 2 above)
Comment (SamirTalwar) [reply packages/server/src/index.ts]: In that case, please handle potential exceptions instead of just deleting the TODO.

File: packages/server/src/index.ts (hunk 1 above)
Comment (SamirTalwar) [reply packages/server/src/index.ts]: Maybe squash before merging to avoid complication.

File: packages/server/src/index.ts [hunk 3]
```diff
@@ -60,15 +61,21 @@ export const start = async (
...
+      try {
+        const matches = await checkMatch(filename);
+        if (matches) {
+          // We need to do this to avoid compiling and pushing `filename` at the
+          // same time as `componentsFile`.
+          await (bundler as any).onChange(componentsFile);
+        }
+      } catch (e) {
+        // tslint:disable-next-line:no-console
+        console.warn("Error handling file change\n", filename, e);
```
Comment (SamirTalwar) [inline packages/server/src/index.ts]: You need a newline before the exception. Perhaps:

```typescript
        console.warn("Error handling change to file:", filename, "\n", e);
```
